platform/linux-dpdk/odp_timer.c

```
@@ -941,6 +1132,57 @@ int odp_timer_cancel(odp_timer_t timer_hdl, odp_event_t *tmo_ev)
 	return 0;
 }
 
+int odp_timer_periodic_cancel(odp_timer_t timer_hdl)
+{
+	odp_timeout_hdr_t *timeout_hdr;
+	timer_entry_t *timer;
+	odp_event_t event;
+	odp_queue_t queue;
+
+	if (odp_unlikely(timer_hdl == ODP_TIMER_INVALID)) {
+		ODP_ERR("Bad timer pool handle\n");
+		return -1;
+	}
+	timer = timer_from_hdl(timer_hdl);
+
+	if (odp_unlikely(!timer->periodic)) {
+		ODP_ERR("Not a periodic timer\n");
+		return -1;
+	}
+
+	odp_ticketlock_lock(&timer->lock);
+
+	if (odp_unlikely(timer->state < TICKING)) {
+		ODP_ERR("Timer not running\n");
+		odp_ticketlock_unlock(&timer->lock);
+		return -1;
+	}
+
+	if (odp_unlikely(timer_global->ops.stop(&timer->rte_timer))) {
+		/* Another core runs timer callback function. */
+		odp_ticketlock_unlock(&timer->lock);
+		return -1;
```

Instead of returning failure in case of stop not possible, cancel could always succeed and stop could be done in ack() calls (until succeeds).

---

Added retry loop in V3.